x/distribution/keeper/delegation.go

```diff
@@ -111,6 +121,12 @@ func (k Keeper) withdrawDelegationRewards(ctx sdk.Context, val sdk.Validator, de
 	// end current period and calculate rewards
 	endingPeriod := k.incrementValidatorPeriod(ctx, val)
 	rewards := k.calculateDelegationRewards(ctx, val, del, endingPeriod)
+	outstanding := k.GetValidatorOutstandingRewards(ctx, del.GetValidatorAddr())
+
+	// defensive edge case may happen on the very final digits
+	// of the decCoins due to operation order of the distribution mechanism.
+	// TODO log if rewards is reduced in this step
+	rewards = sdk.MinSet(rewards, outstanding)
```

Comment:
Despite the fix above? I think this defensive check is OK, but we should definitely log here.

Reply:
logger added